synapse/handlers/federation.py

```diff
@@ -478,6 +480,18 @@ async def do_invite_join(
                 partial_state=ret.partial_state,
             )
 
+            if ret.partial_state:
+                # kick off the process of asynchronously fixing up the state for this
+                # room
+                #
+                # TODO(faster_joins): pick this up again on restart
```

Comment:
I'm guessing we need to persist more info to the database and then ensure that starts up again on restart?

Reply:
I'm not even sure we ned any extra info in the database - all we need is a list of room ids, and we have that.

So yes, it's just a matter of starting it up again on restart.

Reply:
Fair! Was mostly just curious, not asking for it to be fixed yet! 👍